Forta-Bot-Challenge-4/COMP01/src/agent.ts

```diff
@@ -0,0 +1,53 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
+import { SUPPLY_EVENT_SIGNATURE } from "./constants";
+
+import { amountOverThreshold } from "./thresholdCache/thresholdCache";
+
+export function provideHandleTransaction(userAddress: String): HandleTransaction {
+  return async function handleTransaction(txEvent: TransactionEvent) {
+    const findings: Finding[] = [];
+
+    const swapTxs = txEvent.filterLog(SUPPLY_EVENT_SIGNATURE);
+
+    const timestamp = txEvent.timestamp;
+
+    // Iterate through transactions
+    for (const tx of swapTxs) {
+      try {
+        const [from, dst, amount] = tx.args;
+
+        const fromAddress = from;
+
+        // Call isOverThreshold with the timestamp
+        const result = await amountOverThreshold(fromAddress, amount, timestamp);
```

Comment:
```suggestion
let result;
try {
        result = await amountOverThreshold(fromAddress, amount, timestamp);
} catch (e) { ... }
```
It seems the `try/catch` could be around the call to `amountOverThreshold` instead of the current chunk. My above recommendation may need to be further adjust, but hopefully it gets across what I mean.